java/testing/testing-ii/testing-private-methods-and-fields.md

```diff
@@ -36,12 +33,16 @@ User john = new User("John");
 // which has a single argumet of type String
 // "new Class[] { String.class }""
 Method toTest = User
+  .getClass
   .getDeclaredMethod("checkPassHash",
     new Class[] { String.class });
 // make it accessible
 toTest.setAccessible(true);
 // call "checkPassHash()" method on john
-toTest.invoke(john);
+toTest.invoke(
```

Comment:
none of these will work if you don't catch the exceptions

see https://repl.it/@Loopiezlol2/java-reflection

Reply:
in this snippet, you can either pass `new Class[] { String.class });` as arg, or build `arg` above. I feel like the latter is more explicit

Reply:
so I should add this at the of the code block?
```
Class[] arg = new Class[1];  
arg[0] = String.class;

User john = ...
...
```

Reply:
I think that's easier to understand yeah. maybe add comments next to each line saying what we do

Reply:
also add the error handling code as it will not run otherwise; try to mention this fact as well